### DISCUSSION TITLE
Component specific async things

### DISCUSSION THREAD

**Discussion by yatesco** [ivanceras/sauron#68] [Q&A] [answered]
Hi!

Where would I put (typically async) logic that needs to exist only whilst a particular component is alive? I notice that an `App` has an `init` method, but a component doesn't. 

In the react world, I would set up some timers which sync with a server when a particular component was open by starting the timer on the `componentWillMount` and `componentWillUnmount` lifecycle events.

A concrete use-case might be that I want to tail the server log, so I want a timer refreshing the page every X seconds, but only whilst the "Server Log" component is activated.

How would I achieve the same here?

Thanks!

**Comment by ivanceras** [comment, marked as answer]
I haven't put much thought about the init method being in the component. This could also be achieved using the `on_mount` event, however, there is no `on_unmount` event. The yew framework would probably suit your needs much better, since it follows react's design pattern.

**Reply by yatesco** [reply]
thanks @ivanceras - I just *detest* the jsx/rsx style and much prefer the builder pattern :-)
